wiring/inc/spark_wiring_udp.h

```diff
@@ -73,6 +73,22 @@ class UDP : public Stream, public Printable {
          */
         virtual size_t printTo(Print& p) const;
 
+	/*
+	 * Join a multicast address for all UDP sockets. This will allow reception of multicast packets
+	 * sent to the given address for on UDP sockets which have bound the port to which the multicast
+	 * packet was sent.
+	 * @param addr IP multicast address to join
+	 * @return Return the result of the join operation
+	 */
+	static int joinMulticast(const IPAddress& ip);
```

Comment:
If begin() is called first before joining a multicast group, then we could make this an instance method so we have the associated network interface.